### PR TITLE
Add API docs to class.go

### PR DESCRIPTION
**Pull request by hachi8833**

### CODE REVIEW

File: vm/class.go [hunk 1]
```diff
@@ -941,6 +984,39 @@ func builtinClassCommonInstanceMethods() []*BuiltinMethodObject {
...
+		// while the code is executing, which allows the code access to the receiver's
+		// instance variables and private methods.
+		//
+		// No other arguments can be taken.
+		//
+		// ```ruby
+		// string = "String"
+		// string.instance_eval do
+		//   def new_method
+		//		 self.reverse
```
Comment (st0012) [inline vm/class.go]: can you fix the indentation here?

File: vm/class.go [hunk 2]
```diff
@@ -941,6 +984,39 @@ func builtinClassCommonInstanceMethods() []*BuiltinMethodObject {
...
+		//   end
+		// end
+		// string.new_method  #=> "gnirtS"
```
Comment (st0012) [inline vm/class.go]: is this on purpose?

Comment (hachi8833) [reply vm/class.go]: Just a misindentation. I'll fix them.

File: vm/class.go (hunk 1 above)
Comment (hachi8833) [reply vm/class.go]: Just a misindentation. I'll fix them

**Pull request merged**